pub/lib/dependabot/pub/helpers.rb

```diff
@@ -112,6 +115,21 @@ def to_dependency(json)
         Dependency.new(**params)
       end
 
+      # expects "auto" to already have been resolved to one of the other
+      # strategies.
+      def constraint_field_from_update_strategy(requirements_update_strategy)
+        case requirements_update_strategy
+        when nil
+          "constraint"
```

Comment:
This isn't a field anymore. I think `auto` should be `"constraintBumpedIfNeeded"`

Reply:
This is. because we overload this to also parse the "dependency_services list" output. I have split this function in two one for each case. Hope that is clearer.